src/lint.cpp

```diff
@@ -312,6 +312,12 @@ void linter::declare_variable(scope &scope, identifier name, variable_kind kind,
           this->error_reporter_->report(
               error_variable_used_before_declaration{used_var.name, name});
           break;
+        case used_variable_kind::use_and_assignment:
+          this->report_error_if_assignment_is_illegal(
+              declared, used_var.name,
+              /*is_assigned_before_declaration=*/true,
+              /* is_used_before_declaration=*/true);
```

Comment:
Nit: Remove extra space in comment:
```suggestion
              /*is_used_before_declaration=*/true);
```